tests/CoriolisTest.cpp

```diff
@@ -0,0 +1,104 @@
+#include "Tree30Dof.h"
+
+#define BOOST_TEST_MODULE Coriolis
+#include <boost/test/unit_test.hpp>
+
+#include <RBDyn/Coriolis.h>
+#include <RBDyn/EulerIntegration.h>
+#include <RBDyn/FD.h>
+#include <RBDyn/FK.h>
+#include <RBDyn/FV.h>
+
+#include <iostream>
+
+BOOST_AUTO_TEST_CASE(CoriolisTest)
+{
+	std::srand(133757348);
+
+	constexpr double TOL = 1e-6;
+	constexpr double BIGTOL = 50 * TOL;
+
+	rbd::MultiBody mb;
+	rbd::MultiBodyConfig mbc;
+	rbd::MultiBodyGraph mbg;
+
+	std::tie(mb, mbc, mbg) = makeTree30Dof(false);
+
+	rbd::MultiBody mbt;
+	rbd::MultiBodyConfig mbtc;
+	rbd::MultiBodyGraph mbtg;
+
+	std::tie(mbt, mbtc, mbtg) = makeTree30Dof(false);
+
+	const static int ROUNDS = 1000;
+
+	for (int i = 0; i < ROUNDS; ++i)
+	{
+		mbc.zero(mb);
+		mbtc.zero(mbt);
+
+		Eigen::VectorXd q = Eigen::VectorXd::Random(mb.nrParams());
+		mbc.q = rbd::sVectorToParam(mb, q);
+
+		for (auto &q : mbc.q)
+		{
+			if (q.size() == 7)
+			{
+				Eigen::Vector3d axis =
+				    Eigen::Vector3d::Random();
+				Eigen::AngleAxisd aa(0.5, axis / axis.norm());
+				Eigen::Quaterniond qd(aa);
+				q[0] = qd.w();
+				q[1] = qd.x();
+				q[2] = qd.y();
+				q[3] = qd.z();
+			}
+		}
+
+		mbtc.q = mbc.q;
+
+		rbd::forwardKinematics(mb, mbc);
+		rbd::forwardVelocity(mb, mbc);
+
+		rbd::forwardKinematics(mbt, mbtc);
+		rbd::forwardVelocity(mbt, mbtc);
+
+		rbd::ForwardDynamics fd(mbt);
```

Comment:
You can keep this object out of the loop

Reply:
Done